app/models/Contribution.java

```diff
@@ -1568,4 +1577,69 @@ public void setSourceUrl(String sourceUrl){
     public void setCustomFieldValues(List<CustomFieldValue> customFieldValues) {
         this.customFieldValues = customFieldValues;
     }
+
+    private static double[] polygonCenter(double[]... polygonPoints) {
```

Comment:
This is probably something we will re-use in many places. I would suggest we move this either to `app/models/location/Location.java` or to `app/utils/[SomeNewClassWithGEOUtilities]`